errors don't appear on save with go 1.13

### What did you do? (required: The issue will be **closed** when not provided)
I update to go 1.13.

### What did you expect to happen?
Everything to work fine.

### What happened instead?
Vim is not showing errors any more

### Configuration (**MUST** fill this out):

#### vim-go version:
1.20-1

####  `vimrc` you used to reproduce (use a *minimal* vimrc with other plugins disabled; do not link to a 2,000 line vimrc):
<details><summary>vimrc</summary><br><pre>
source $VIMRUNTIME/defaults.vim
set mouse=""
set nu
set laststatus=2
set t_Co=256
set wildmode=longest,list
let g:powerline_pycmd = 'py3'
nnoremap <C-Up> <C-W><C-K>
nnoremap <C-Down> <C-W><C-J>
nnoremap <C-Right> <C-W><C-L>
nnoremap <C-Left> <C-W><C-H>
nnoremap <F5> :terminal % <CR>

set splitbelow
set splitright
set shiftwidth=2
set expandtab
set softtabstop=2
let g:indentLine_color_term = 239
let g:ansible_unindent_after_newline = 1
let g:go_auto_sameids = 1

let g:go_highlight_array_whitespace_error = 1
let g:go_highlight_chan_whitespace_error = 1
let g:go_highlight_extra_types = 1
let g:go_highlight_space_tab_error = 1
let g:go_highlight_trailing_whitespace_error = 0
let g:go_highlight_operators = 1
let g:go_highlight_functions = 1
let g:go_highlight_function_parameters = 1
let g:go_highlight_function_calls = 1
let g:go_highlight_types = 1
let g:go_highlight_fields = 1
let g:go_highlight_build_constraints = 1
let g:go_highlight_generate_tags = 1
let g:go_highlight_string_spellcheck = 1
let g:go_highlight_format_strings = 1
let g:go_highlight_variable_declarations = 1
let g:go_highlight_variable_assignments = 1

colorscheme pablo
hi Folded ctermbg=232 ctermfg=33
set foldmethod=syntax
set foldnestmax=1
let g:go_fmt_experimental = 1

</pre></details>

#### Vim version (first three lines from `:version`):
<!-- :version -->
8.1.1968-1

#### Go version (`go version`):
<!-- go version -->
go1.13

#### Go environment
<details><summary><code>go env</code> Output:</summary><br><pre>
<!-- go env -->
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/gunix/.cache/go-build"
GOENV="/home/gunix/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/gunix/.go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/lib/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="0"
GOMOD="/home/gunix/git/gun1x/spiderweber/spider/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build214595135=/tmp/go-build -gno-record-gcc-switches"

</pre></details>

@gun1x  I've taken another look at this.

The error you were seeing on save was never from `gofmt` from what I can tell. The _only_ way that I can duplicate showing compilation errors on save is if `g:go_metalinter_autosave` is enabled, `g:go_metalinter_command` is set to `gometalinter`, and Go 1.12.9 or lower is installed.

After a lot of digging, I tracked the problem you reported down to a change in `go vet`'s output. By default, vim-go calls `gometalinter` with the `vet` linter when `g:go_metalinter_autosave` is set.
Prior to Go 1.13, `go vet`'s output (and consequently `gometalinter`'s) looked something like this: `./main.go:6:2: undefined: foo`. With Go 1.13, though, `go vet`'s output includes a prefix before the file location`: `vet: ./main.go:6:2: undeclared name: foo`, which breaks `gometalinter`'s expectations.

Since `gometalinter` is archived and deprecated by its owner, there's not much we can do here; `gometalinter` is effectively EOL, and the next release of vim-go will no longer use it for the default `g:go_metalinter_command`.

The one bit I'm confused about though, is that I don't see `let g:go_metalinter_autosave=1` in your vimrc. 🤔

--- a/autoload/go/lint.vim
+++ b/autoload/go/lint.vim
@@ -36,7 +36,7 @@ function! go#lint#Gometa(bang, autosave, ...) abort
       " Include only messages for the active buffer for autosave.
       let include = [printf('--include=^%s:.*$', fnamemodify(expand('%:p'), ":."))]
       if go#util#has_job()
-        let include = [printf('--include=^%s:.*$', expand('%:p:t'))]
+        let include = [printf('--include=^(vet: \.[\\/])?%s:.*$', expand('%:p:t'))]
       endif
       let cmd += include
     elseif l:metalinter == "golangci-lint"
@@ -232,6 +232,9 @@ function! s:lint_job(args, bang, autosave)
 
   if a:autosave
     let l:opts.for = "GoMetaLinterAutoSave"
+    " s:metalinterautosavecomplete is really only needed for golangci-lint,
+    " but s:lint_job doesn't know whether golangci-lint is being called or
+    " whether gometalinter is being called, so hook it up in either case.
     let l:opts.complete = funcref('s:metalinterautosavecomplete', [expand('%:p:t')])
   endif
 
@@ -287,10 +290,14 @@ function! s:metalinterautosavecomplete(filepath, job, exit_code, messages)
     return
   endif
 
-  let l:file = expand('%:p:t')
   let l:idx = len(a:messages) - 1
   while l:idx >= 0
-    if a:messages[l:idx] !~# '^' . a:filepath . ':'
+    " Go 1.13 changed how go vet output is formatted by prepending a leading
+    " 'vet :', so account for that, too. This function is really needed for
+    " gometalinter at all, so the check for Go 1.13's go vet output shouldn't
+    " be neeeded, but s:lint_job hooks this up even when the
+    " g:go_metalinter_command is golangci-lint.
+    if a:messages[l:idx] !~# '^' . a:filepath . ':' && a:messages[l:idx] !~# '^vet: \.[\\/]' . a:filepath . ':'
       call remove(a:messages, l:idx)
     endif
     let l:idx -= 1

